hs_core/views/utils.py

```diff
@@ -1093,30 +1093,55 @@ def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original,
         raise ValidationError("Unzipping of file is not allowed for a published resource.")
     istorage = resource.get_irods_storage()
     zip_with_full_path = os.path.join(resource.root_path, zip_with_rel_path)
+    if not istorage.exists(zip_with_full_path):
+        raise ValidationError("Zip file was not found.")
+
     if not resource.supports_unzip(zip_with_rel_path):
         raise ValidationError("Unzipping of this file is not supported.")
 
-    unzip_path = None
+    unzip_to_folder_path = ''
+    unzip_path_temp = ''
     try:
+        # unzip to a temporary folder first to validate contents of the zip file
+        unzip_path_temp = istorage.unzip(zip_with_full_path, unzipped_folder=uuid4().hex)
+
+        # validate files in the zip file - checking for banned characters
+        unzipped_files = listfiles_recursively(istorage, unzip_path_temp)
+        for file_in_zip in unzipped_files:
+            file_in_zip = os.path.basename(file_in_zip)
+            if not ResourceFile.is_filename_valid(file_in_zip):
+                log_msg = f"Failed to unzip. Zip file ({zip_with_full_path}) has file with name that contains " \
+                          f"one or more prohibited characters."
+                logger.error(log_msg)
+                err_msg = "Zip file has file with name that contains one or more prohibited characters."
+                raise SuspiciousFileOperation(err_msg)
```

Comment:
I guess I wonder if we should be raising a `SuspiciousFileOperation` or perhaps some other type of exception? Like a [Validationerror](https://docs.djangoproject.com/en/4.1/ref/exceptions/#validationerror)? 

https://docs.djangoproject.com/en/4.1/ref/exceptions/#suspiciousoperation
The docs say

> The [SuspiciousOperation](https://docs.djangoproject.com/en/4.1/ref/exceptions/#django.core.exceptions.SuspiciousOperation) exception is raised when a user has performed an operation that should be considered suspicious from a security perspective, such as tampering with a session cookie

I guess the filename characters doesn't feel particularly "suspicious" to me, but perhaps this is the best exception type we have available...

Reply:
Yeah initially I was also bit hesitant to use the exception class `SuspiciousFileOperation`. Then I noticed we are already using in similar validation of file path and Django also using it in the get_valid_filename() (https://docs.djangoproject.com/en/3.2/_modules/django/utils/text/). May be we can use a more readable custom exception like `InvalidPathNameError`.

Reply:
`InvalidPathNameError` is fine with me! Or sticking with `SuspiciosFileOperation` to be consistent is fine too...